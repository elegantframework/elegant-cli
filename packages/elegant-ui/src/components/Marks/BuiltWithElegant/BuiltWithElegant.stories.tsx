import type { Meta, StoryObj } from '@storybook/react';
import BuiltWithElegant from './BuiltWithElegant';

const meta: Meta<typeof BuiltWithElegant> = {
  title: 'Core/Marks/Built with Elegant',
  component: BuiltWithElegant,
};

export default meta;
type Story = StoryObj<typeof BuiltWithElegant>;

export const Primary: Story = {
  args: {},
};