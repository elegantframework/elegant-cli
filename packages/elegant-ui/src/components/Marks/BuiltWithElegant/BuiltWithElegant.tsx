import React from "react";
import clsx from 'clsx';

export interface Props {
    /**
     * The url css text color classes.
     */
    linkColor?: string;
    /**
     * A copyright company name to be included in the footer.
     */
    copyright?: string;
};

/**
 * The Built with Elegant mark to be used in footers.
 * @returns An html element to be used in Elegant footers.
 */
const BuiltWithElegant = ({
    linkColor="hover:text-primary-500 dark:hover:text-primary-400",
    copyright="Elegant, Inc."
}: Props) => {
    return(
        <div className="mb-6 sm:mb-0 sm:flex">
            <p>Copyright &copy; {new Date().getFullYear()} {copyright}</p>
            <p className="sm:ml-4 sm:pl-4 sm:border-l sm:border-slate-200 dark:sm:border-slate-200/5">
                Built with <a href='http://elegantframework.com/' 
                                aria-label='Built with the Elegant framework'
                                className={clsx('font-semibold', linkColor)}
                            >
                                Elegant
                            </a>.
            </p>
        </div>
    );
}

export default BuiltWithElegant;