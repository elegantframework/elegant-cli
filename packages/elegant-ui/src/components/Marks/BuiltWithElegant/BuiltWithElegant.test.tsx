import React from "react";
import renderer from 'react-test-renderer';
import BuiltWithElegant from "./BuiltWithElegant";

describe("Built with Elegant component", () => {
    it('renders a built with Elegant component', () => {
        const element = renderer
        .create(<BuiltWithElegant />)
        .toJSON();
        expect(element).toMatchSnapshot();
    });
});