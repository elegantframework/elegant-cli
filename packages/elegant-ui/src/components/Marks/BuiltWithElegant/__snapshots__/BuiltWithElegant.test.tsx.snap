// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Built with Elegant component renders a built with Elegant component 1`] = `
<div
  className="mb-6 sm:mb-0 sm:flex"
>
  <p>
    Copyright Â© 
    2024
     
    Elegant, Inc.
  </p>
  <p
    className="sm:ml-4 sm:pl-4 sm:border-l sm:border-slate-200 dark:sm:border-slate-200/5"
  >
    Built with 
    <a
      aria-label="Built with the Elegant framework"
      className="font-semibold hover:text-primary-500 dark:hover:text-primary-400"
      href="http://elegantframework.com/"
    >
      Elegant
    </a>
    .
  </p>
</div>
`;
